engine/table/src/main/java/io/deephaven/engine/table/impl/replay/ReplayTable.java

```diff
@@ -4,69 +4,83 @@
 package io.deephaven.engine.table.impl.replay;
 
 import io.deephaven.base.verify.Require;
+import io.deephaven.engine.rowset.*;
+import io.deephaven.engine.table.impl.sources.ReinterpretUtils;
 import io.deephaven.time.DateTime;
 import io.deephaven.engine.table.impl.QueryTable;
 import io.deephaven.engine.table.ColumnSource;
-import io.deephaven.engine.rowset.RowSet;
-import io.deephaven.engine.rowset.RowSetBuilderRandom;
-import io.deephaven.engine.rowset.RowSetFactory;
+import io.deephaven.util.QueryConstants;
+import org.jetbrains.annotations.NotNull;
 
 import java.util.Map;
 
 public class ReplayTable extends QueryTable implements Runnable {
 
+    private final Replayer replayer;
+    private final ColumnSource<Long> nanoTimeSource;
+    private final RowSet.Iterator rowSetIterator;
 
-    private final RowSet.Iterator indexIterator;
-    private long curr;
-    private final ColumnSource<DateTime> timeSource;
+    private long nextRowKey = RowSequence.NULL_ROW_KEY;
+    private long nextTimeNanos = QueryConstants.NULL_LONG;
     private boolean done;
-    private final Replayer replayer;
 
-    public ReplayTable(RowSet rowSet, Map<String, ? extends ColumnSource<?>> result, String timeColumn,
-            Replayer replayer) {
-        super(RowSetFactory.empty().toTracking(), result);
-        Require.requirement(replayer != null, "replayer != null");
-        // noinspection unchecked
-        replayer.registerTimeSource(rowSet, (ColumnSource<DateTime>) result.get(timeColumn));
+    public ReplayTable(
+            @NotNull final RowSet rowSet,
+            @NotNull final Map<String, ? extends ColumnSource<?>> columns,
+            @NotNull final String timeColumn,
+            @NotNull final Replayer replayer) {
+        super(RowSetFactory.empty().toTracking(), columns);
+        this.replayer = Require.neqNull(replayer, "replayer");
+        // NB: This will behave incorrectly if our row set or any data in columns can change. Our source table *must*
+        // be static. We also seem to be assuming that timeSource has no null values in rowSet. It would be nice to use
+        // a column iterator for this, but that would upset unit tests by keeping pooled chunks across cycles.
```

Comment:
1) We don't want initials in comments or code.
2) The comment implies that we have a requirement that the input table is static.  Can we assert that here from the column sources?
3) @rcaudy How big of a deal is the column iterator comment?
4) See other comments about null times.  We should raise an error on null times.
5) I don't think we are asserting the time order.  We should also raise an error on out-of-order times, if we do not support them.

Reply:
NB stands for Nota Bene. There are no initials here.

We can't tell anything about whether the row set or column sources can change from here. Would need the Table. The calling code handles it for now.

Reply:
The column iterator comment is just an implementation note for myself or other devs. No big deal at all.